src/components/outline/DraggableTree.tsx

```diff
@@ -178,8 +178,12 @@ export const DraggableTree: React.FC<DraggableTreeProps> = ({
     accept: 'outline',
     canDrop(item: TreeItem, monitor) {
       if (
-        tree.node.type === schema.nodes.bibliography_section ||
-        item.node.type === schema.nodes.bibliography_section
+        item.parent &&
```

Comment:
Ahmad, let's make this condition easier to read and understand.

I would suggest adding a helper function:

```
  const isAbstractOrBackmatter = (node: ManuscriptNode) => {
    return (
      node.type === schema.nodes.abstracts ||
      node.type === schema.nodes.backmatter
    )
  }
  ```
  
 Then use it like this:
  
  ```
  if (!tree.parent || !item.parent) {
        return false
      }
        const isInvalidDrop =
        isAbstractOrBackmatter(item.parent) ||
        isAbstractOrBackmatter(tree.parent)

      if (isInvalidDrop) {
        return false
      }
  ```
  
Also, we can remove the condition `if (!tree.parent)` in line 193.

Would that work for you?

Reply:
updated

Reply:
Sounds good. You can remove the condition` if (!tree.parent)` on line 194 as it will not be reached.

Reply:
hmm, will reach that location! updated to `!tree.parent || isAbstractOrBackmatter(item) || isAbstractOrBackmatter(tree)`

Reply:
I see how it will reach it. Sorry for the confusion! I would stick with the previous separate conditions. Sorry again!

Reply:
no worry that is ok, reverted